torchtext/prototype/models/t5/model.py

```diff
@@ -136,12 +131,24 @@ def __init__(
             for p in self.parameters():
                 p.requires_grad = False
 
+    def prepare_inputs_for_generation(self, input_ids, encoder_outputs, **kwargs):
+        return {"decoder_tokens": input_ids, "encoder_outputs": encoder_outputs}
+
+    def get_encoder(self) -> T5Encoder:
+        return self.encoder
+
+    def get_decoder(self) -> T5Decoder:
+        return self.decoder
+
     def forward(
         self,
-        encoder_tokens: Tensor,
+        encoder_tokens: Optional[Tensor] = None,
```

Comment:
Don't have to include `encoder_tokens` if `encoder_outputs` are already provided.